toolchain/sem_ir/ids.h

```diff
@@ -112,6 +112,36 @@ struct BoolValue : public IndexBase, public Printable<BoolValue> {
 constexpr BoolValue BoolValue::False = BoolValue(0);
 constexpr BoolValue BoolValue::True = BoolValue(1);
 
+// The ID of a name. A name is either a string or a special name such as
+// `self`, or eventually `Self` or `base`.
+struct NameId : public IndexBase, public Printable<NameId> {
+  // An explicitly invalid ID.
+  static const NameId Invalid;
+  // The name of `self`.
+  static const NameId SelfValue;
+  // The name of the return slot in a function.
+  static const NameId ReturnSlot;
```

Comment:
Should the constraints that this creates for potential negative values of StringId (or IdentifierId) be documented?

Reply:
Done.